lib/sqlalchemy/dialects/postgresql/psycopg2.py

```diff
@@ -551,13 +589,41 @@ def _log_notices(self, cursor):
 
 
 class PGCompiler_psycopg2(PGCompiler):
-    pass
+    execute_values_insert_template = None
+
+    # Override SQLCompiler.generate_values_placeholders_str - enable use of
+    # psycopg2 execute_values()
+    def generate_values_placeholders_str(
+            self, crud_params, returning_clause_exists):
+
+        # Currently not using psycopg2.execute_values() when there's a
+        # returning clause; need to add support
+        # for receiving multiple return values from insert query
+        # Note: self.inline is true iff there are multiple parameters sets to
+        # the query
+        if self.inline and not returning_clause_exists and \
```

Comment:
**mike bayer** (zzzeek) wrote:

is it guaranteed this will be an executemany()?

Reply:
Do you mean when self.inline is true? This is what I deduced from the code, I hope that's right